### ISSUE TITLE
False positive in Directive class name in Angular 5

### ISSUE THREAD

**Issue by mariohmol** [palantir/tslint#3739]
### Bug Report

- __TSLint version__: ^5.9.1
- __TypeScript version__: 2.4.2
- __Running TSLint via__: VSCODE

#### TypeScript code being linted

```ts
export class SelectItemDirective implements OnInit, OnChanges {
```

Error i'm getting

````
[tslint] The name of the class SelectItemDirective should end with the suffix Component (https://angular.io/styleguide#style-02-03) (component-class-suffix)
class SelectItemDirective 
```
The name of the file is `selectitem.directive.ts`

with `tslint.json` configuration:

```json
{
  "rulesDirectory": [
    "node_modules/codelyzer"
  ],
  "rules": {
    "arrow-return-shorthand": true,
    "callable-types": true,
    "class-name": true,
    "comment-format": [
      true,
      "check-space"
    ],
    "curly": true,
    "eofline": true,
    "forin": true,
    "import-blacklist": [
      true,
      "rxjs"
    ],
    "import-spacing": true,
    "indent": [
      true,
      "spaces"
    ],
    "interface-over-type-literal": true,
    "label-position": true,
    "max-line-length": [
      true,
      140
    ],
    "member-access": false,
    "member-ordering": [
      true,
      {
        "order": [
          "static-field",
          "instance-field",
          "static-method",
          "instance-method"
        ]
      }
    ],
    "no-arg": true,
    "no-bitwise": true,
    "no-console": [
      true,
      "debug",
      "info",
      "time",
      "timeEnd",
      "trace"
    ],
    "no-construct": true,
    "no-debugger": true,
    "no-duplicate-super": true,
    "no-empty": false,
    "no-empty-interface": true,
    "no-eval": true,
    "no-inferrable-types": [
      true,
      "ignore-params"
    ],
    "no-misused-new": true,
    "no-non-null-assertion": true,
    "no-shadowed-variable": true,
    "no-string-literal": false,
    "no-string-throw": true,
    "no-switch-case-fall-through": true,
    "no-trailing-whitespace": true,
    "no-unnecessary-initializer": true,
    "no-unused-expression": true,
    "no-use-before-declare": true,
    "no-var-keyword": true,
    "object-literal-sort-keys": false,
    "one-line": [
      true,
      "check-open-brace",
      "check-catch",
      "check-else",
      "check-whitespace"
    ],
    "prefer-const": true,
    "quotemark": [
      true,
      "single"
    ],
    "radix": true,
    "semicolon": [
      true,
      "always",
      "ignore-interfaces"
    ],
    "triple-equals": [
      true,
      "allow-null-check"
    ],
    "typedef-whitespace": [
      true,
      {
        "call-signature": "nospace",
        "index-signature": "nospace",
        "parameter": "nospace",
        "property-declaration": "nospace",
        "variable-declaration": "nospace"
      }
    ],
    "typeof-compare": true,
    "unified-signatures": true,
    "variable-name": false,
    "whitespace": [
      true,
      "check-branch",
      "check-decl",
      "check-operator",
      "check-separator",
      "check-type"
    ],
    "directive-selector": [
      true,
      "attribute",
      "app",
      "camelCase"
    ],
    "component-selector": [
      true,
      "element",
      "app",
      "kebab-case"
    ],
    "use-input-property-decorator": true,
    "use-output-property-decorator": true,
    "use-host-property-decorator": true,
    "no-input-rename": true,
    "no-output-rename": true,
    "use-life-cycle-interface": true,
    "use-pipe-transform-interface": true,
    "component-class-suffix": true,
    "directive-class-suffix": true
    // "no-access-missing-member": true,
    // "templates-use-public": true,
    // "invoke-injectable": true
  }
}

```

#### Actual behavior

Lint is giving and error asking me to change the class name to Component

#### Expected behavior

It points to an angular url https://angular.io/guide/styleguide#style-02-03, and there its specified about Directive

I tryed to find some issue that talks about it but i couldnt, so i opnened this bug.

Thanks

**Comment by aervin** [comment]
Hey @mariohmol, you may want to move this to [codelyzer](https://github.com/mgechev/codelyzer). I don't think `component-class-suffix` is a core TSLint rule.

**Comment by mariohmol** [comment]
Thanks, opened there https://github.com/mgechev/codelyzer/issues/522
